adapter/utils/src/test/java/GeodeticUtilsTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2020. MobilityData IO.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import org.junit.jupiter.api.Test;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.ShapePoint;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceCalculationUtils;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceUnit;
+import org.mobilitydata.gtfsvalidator.utils.GeodeticUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class GeodeticUtilsTest {
+    private static final float MONTREAL_LATITUDE = 45.5017f;
+    private static final float MONTREAL_LONGITUDE = -73.5673f;
+    private static final float NEW_YORK_LATITUDE = 40.7128f;
+    private static final float NEW_YORK_LONGITUDE = -74.0060f;
+    private static final float POINTE_A_PITRE_LATITUDE = 16.2650f;
+    private static final float POINTE_A_PITRE_LONGITUDE = -61.5510f;
+    private static final float FORT_DE_FRANCE_LATITUDE = 14.6161f;
+    private static final float FORT_DE_FRANCE_LONGITUDE = -61.0588f;
+
+    @Test
+    void distanceBetweenSamePointShouldBeZero(){
+        assertEquals(0, new GeodeticUtils().distanceBetweenTwoPoints(
```

Comment:
Generally, you shouldn't need to instantiate a utility class. Unless this is part of the clean architecture guidelines. Utility methods should be static.

Reply:
True, however here we implement an interface which does not allow us to declare methods as `static` if we want to follow the clean architecture principles: if the method is declared static within the interface, we have to implement it there. Which means that external libraries will be present within the `usecase` layer.

Reply:
Yeah I had the same hunch yesterday but as @lionel-nj mentioned it seems we don't have a choice given we use Java and want to declare interfaces in our domain/use case layer and relegate the implementation to the adapter one. That being said I think that would be a good use case to implement a Singleton. https://www.geeksforgeeks.org/singleton-class-java/